sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java

```diff
@@ -0,0 +1,386 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.CosmosPagedFluxOptions;
+import com.azure.cosmos.implementation.changefeed.implementation.ChangeFeedMode;
+import com.azure.cosmos.implementation.changefeed.implementation.ChangeFeedStartFromInternal;
+import com.azure.cosmos.implementation.changefeed.implementation.ChangeFeedState;
+import com.azure.cosmos.implementation.feedranges.FeedRangeContinuation;
+import com.azure.cosmos.implementation.feedranges.FeedRangeInternal;
+import com.azure.cosmos.implementation.query.CompositeContinuationToken;
+import com.azure.cosmos.util.Beta;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+@Beta(value = Beta.SinceVersion.WHATEVER_NEW_VERSION, warningText =
+    Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public final class CosmosChangeFeedRequestOptions {
+    private static final int DEFAULT_MAX_ITEM_COUNT = 1000;
```

Comment:
Having these settings on the public facing "models" class become binding to the API and hard to change after. IMHO 1000 is a bit high, I will recommend starting with 100 instead.

Reply:
Changed it to 100 - but where I define the default doesn't matter - any change to the default value is a behavior change that would be considered breaking. I'll changed it to 100 (the default in .Net implicitly and the server-side REST api as well)

Reply:
Fixed (next iteration)